### PR TITLE
report better s3Errors

### PR DESCRIPTION
**Pull request by rakeshgm**
use aws/awserr package to report
better errors with correct format

issue: https://github.com/RamenDR/ramen/issues/749
[Bugzilla 2069759](https://bugzilla.redhat.com/show_bug.cgi?id=2069759)

### CODE REVIEW

File: controllers/s3utils.go [hunk 1]
```diff
@@ -375,6 +376,15 @@ func DeleteTypedObjects(s ObjectStorer, keyPrefix, keySuffix string, object inte
 	return s.DeleteObjects(typedKey(keyPrefix, keySuffix, reflect.TypeOf(object)))
 }
 
+func processAwsError(errMsgPrefix string, err error) error {
+	var awsErr awserr.Error
+	if errors.As(err, &awsErr) {
+		return fmt.Errorf("%s: code: %s, message: %s", errMsgPrefix, awsErr.Code(), awsErr.Message())
+	}
+
+	return fmt.Errorf("%s: not an aws error", errMsgPrefix)
```
Comment (BenamarMk) [inline controllers/s3utils.go:385]: Just return `errMsgPrefix`

Comment (rakeshgm) [reply controllers/s3utils.go]: done

**Review by BenamarMk — APPROVED** [review]

**Review by ShyamsundarR — COMMENTED** [review]
LGTM!

Question: Why not handle this in other calls as well like `DeleteBucket`, `CreateBucket`, ...? They could also return malformed errors that would end up in the resource status.

**Review by ShyamsundarR — COMMENTED** [review]
@rakeshgm Please handle these instances as well:

- https://github.com/RamenDR/ramen/blob/fe846a6f4870a2305fa81bbba5460a6208960d08/controllers/s3utils.go#L579
    - We return the err back to higher layers, and should process it like in the other places
- https://github.com/RamenDR/ramen/blob/fe846a6f4870a2305fa81bbba5460a6208960d08/controllers/s3utils.go#L627
    - Here error is returned back to caller, would be good to process it right here and return instead

**Pull request merged**